core/src/types/config.rs

```diff
@@ -44,14 +44,82 @@ pub struct AWSConfig {
     pub artifact_bucket_name: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct AppConfig {
+pub type SharedConfigRef = Arc<RwLock<SharedConfig>>;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SharedConfig {
```

Comment:
Can you help me understand the purpose of this struct a bit more?

I see how it's used for the RepoWorker, but am curious if maybe there's a different way we can initialize the RepoWorker. 

It feels to me like having this shared struct actually adds more code than it saves because of all the places you need to check for it. I'm also a tad bit concerned about lock contention and timing issues as we've now got a lockable inside a lockable, which isn't categorically bad but I could see us deadlocking in the future if we aren't careful.

Reply:
its just for shared fields
I do agree, the double lock mechanism does worry me

Im fine with duplicating fields, and then refactoring the functions to take args instead